pom.xml

```diff
@@ -45,8 +56,8 @@
                 <artifactId>maven-compiler-plugin</artifactId>
                 <version>2.3.2</version>
                 <configuration>
-                    <source>1.6</source>
-                    <target>1.6</target>
+                    <source>1.8</source>
```

Comment:
We do need to support at least 1.7, because most customers haven't moved to 1.8 yet. Supporting 1.6 is optional, because ES itself doesn't have official support for it. (https://www.elastic.co/support/matrix)